src/utilities/product-context.js

```diff
@@ -0,0 +1,50 @@
+import {React, useReducer, createContext, useContext} from 'react';
+import { initialFilter } from './initialFilter';
+import { useInitialProducts } from './useInitialProducts';
+import {getFilteredProducts} from "./getFilteredProducts";
+import { sortFun, fastDeliveryFun, ratingFun, categoryFun, stockFun, priceRangeFun} from './filterFunctions';
+
+const ProductContext = createContext();
+
+const filterFunction = (state, action) => {
+    switch (action.type) {
+        case "LOW-TO-HIGH": return {...state, sorting: action.payload};
+        case "HIGH-TO-LOW": return {...state, sorting: action.payload};
+        case "IN-STOCK": return {...state, inStock: !state.inStock};
+        case "IS-FAST-DELIVERY": return {...state, isFastDelivery: !state.isFastDelivery};
+        case "RATING": return {...state, rating: action.payload};
+        case "PRICE-RANGE": return {...state, priceRange: action.payload};
+        case "CATEGORY": return {...state, 
+            categories: state['categories'].includes(action.payload) 
+            ? state['categories'].filter(category => category !== action.payload) 
+            : state['categories'].concat(action.payload)
+        };
+        case "CLEAR": return initialFilter;
+        default: return state;
+    }
+}
```

Comment:
This could be exported as its own function and imported to context file, just like youve done for other files, making your code look even cleaner and orgnaized

Reply:
Sure I will do it. 👍